### PR TITLE
refactor: Expense Claim Cost Center

### PR DESCRIPTION
**Pull request by jiloysss**
* There is a cost center field added to the table (Expense Claim Detail)
* Changed Script in Expense Claim (python) where the cost center in GL Entry is based on the cost center field on each row (Expense Claim Detail)
* Filtered Cost Center in Expense Claim Detail where it will only fetch child account
* Hid Cost Center in Expense Claim
* Added Deferred Expense Accounts checkbox in Expense Claim Type

Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

### CODE REVIEW

**Comment by jiloysss** [comment]
EXPENSE CLAIM
Added field Cost Center in table

expense_claim

EXPENSE CLAIM TYPE
Added check box Deferred Expense Accounts
If checked, Default Account in table will be filtered with Asset Accounts. Expense Accounts if otherwise

Expense_claim_type

**Review by sammish — APPROVED** [review]

File: erpnext/hr/doctype/expense_claim/expense_claim.js [hunk 1]
```diff
@@ -402,3 +403,10 @@ cur_frm.fields_dict['task'].get_query = function(doc) {
 		}
 	};
 };
+cur_frm.fields_dict["expenses"].grid.get_field("cost_center").get_query = function(){
+	return {
+		filters:{
+			"is_group": "child"
```
Comment (sammish) [inline erpnext/hr/doctype/expense_claim/expense_claim.js:409]: "is_group": 0

File: erpnext/hr/doctype/expense_claim/expense_claim.js [hunk 2]
```diff
@@ -61,6 +61,7 @@ cur_frm.cscript.clear_sanctioned = function(doc) {
 };
 
 cur_frm.cscript.refresh = function(doc) {
+	cur_frm.set_df_property("cost_center", "hidden", true);
```
Comment (nabinhait) [inline erpnext/hr/doctype/expense_claim/expense_claim.js:64]: - Don't hide cost_center field, remove it entirely. And write a patch to update child cost centers based on parent.

- Copy cost center from first row on addition of new row. And on entering cost_center in any row, it should get copied to all rows with blank cost_center.

Refer following code for example:
https://github.com/frappe/erpnext/blob/develop/erpnext/buying/doctype/purchase_order/purchase_order.js#L458
https://github.com/frappe/erpnext/blob/develop/erpnext/buying/doctype/purchase_order/purchase_order.js#L556

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/expense_claim/expense_claim.js (hunk 2 above)
Comment (sammish) [reply erpnext/hr/doctype/expense_claim/expense_claim.js:64]: @nabinhait It has updated kindly check  #19663

Comment (sammish) [reply erpnext/hr/doctype/expense_claim/expense_claim.js:64]: @nabinhait It has updated again #19663 

Please check and verify.

**Pull request closed without merging**